python/rmm/_lib/device_buffer.pyx

```diff
@@ -21,25 +23,41 @@ cdef class DeviceBuffer:
     def __len__(self):
         return self.size
 
+    @property
+    def dtype(self):
```

Comment:
`rmm::device_buffer` is anuntyped memory buffer. So it doesn't make sense to treat it like a typed array.

Reply:
This is basically treating it as `unsigned char` ATM, which behaves similarly to checking type on a `memoryview` of `bytes` in Python (please see below).

```python
In [1]: m = memoryview(b"abc")                                                  

In [2]: m.format  # same as `|u1`
Out[2]: 'B'
```

Reply:
What we're doing in `__cuda_array_interface__` is very much a workaround in my mind at the moment. Python Buffer protocol may have array-like properties, but that seems out of scope for just defining memory buffers, which is the goal of `DeviceBuffer`. If we want a thin array-like wrapper on top of `DeviceBuffer` that can wrap some things from `device_vector` as well then that would make more sense, but then we get into the territory of creating yet another GPU array library.

Reply:
I wouldn't call using `__cuda_array_interface__` a workaround with `DeviceBuffer`. This is the same as supporting [the Python buffer protocol]( https://docs.python.org/3/c-api/buffer.html ) with `bytes` objects.

That said, I do see your point that we might not want `DeviceBuffer` to explicitly expose array properties.